### PR TITLE
[server] Fix A/A RMD missing during repush when chunking is enabled

### PR DESCRIPTION
**Pull request by sixpluszero**
## [server] Fix A/A RMD missing during repush when chunking is enabled
This PR adds:
1. Available RMD info during repush when chunking is enabled
2. In the KIF reducer when chunking is enabled, make sure DELETE message is included when RMD data is present.

## How was this PR tested?
Internal CI, adjusted integration tests and unit tests.
## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

**Review by gaojieliu — COMMENTED** [review]
Can you add some E2E test to validate the repush with RMD?

**Comment by sixpluszero** [comment]
> Can you add some E2E test to validate the repush with RMD?

I think TestActiveActive#testKIFRepushActiveActiveStore has covered the case - including chunking case. However, I think e2e test cannot validate RMD as we don't have easy way to check RMD within integration framework. Do you think it is good to have a unit test for KIF reducer side or it is fine for now?

**Comment by adamxchen** [comment]
> > Can you add some E2E test to validate the repush with RMD?
> 
> I think TestActiveActive#testKIFRepushActiveActiveStore has covered the case - including chunking case. However, I think e2e test cannot validate RMD as we don't have easy way to check RMD within integration framework. Do you think it is good to have a unit test for KIF reducer side or it is fine for now?

I just revisit the code of `TestActiveActive#testKIFRepushActiveActiveStore`. Although we have chunking flag on, the records themselves are not exceeding 1MB. I am not sure the chunking behavior if the record size is less 1MB. Do you know it? If the chunking is still enabled anyway, then we should be good to go!

**Comment by sixpluszero** [comment]
> > > Can you add some E2E test to validate the repush with RMD?
> > 
> > 
> > I think TestActiveActive#testKIFRepushActiveActiveStore has covered the case - including chunking case. However, I think e2e test cannot validate RMD as we don't have easy way to check RMD within integration framework. Do you think it is good to have a unit test for KIF reducer side or it is fine for now?
> 
> I just revisit the code of `TestActiveActive#testKIFRepushActiveActiveStore`. Although we have chunking flag on, the records themselves are not exceeding 1MB. I am not sure the chunking behavior when the record size is less 1MB. A bit concerned if they are treated as regular records, which means chunking-logic isn't exercised at all in these tests. Do you have any insights on this?

Hi Adam, thanks for pointing out. I don't think 1MB is a concern as when chunking is enabled, it always go to a specific method in the KIF reducer logic no matter the size of the value (which it does not know in advance). But still thanks for pointing out. I adjust that integration test logic to cover this change.

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestActiveActiveIngestion.java [hunk 1]
```diff
@@ -379,6 +407,24 @@ private void runSamzaStreamJob(String storeName, Time mockedTime, int numPuts, i
...
 
+  private void produceRecordWithLogicalTimestamp(
+      String storeName,
+      int index,
+      long logicalTimestamp,
+      boolean isDeleteOperation) {
+    Map<String, String> samzaConfig = getSamzaConfig(storeName);
+    VeniceSystemFactory factory = new VeniceSystemFactory();
+    try (
+        VeniceSystemProducer veniceProducer = factory.getClosableProducer("venice", new MapConfig(samzaConfig), null)) {
```
Comment (majisourav99) [inline internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestActiveActiveIngestion.java:418]: can we reuse the producer from existing samza jobs ?

**Comment by majisourav99** [comment]
> 

Still would be good to have messages > 1Mb in an E2E test.

File: internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestActiveActiveIngestion.java (hunk 1 above)
Comment (sixpluszero) [reply internal/venice-test-common/src/integrationtest/java/com/linkedin/venice/endToEnd/TestActiveActiveIngestion.java:418]: Either way is fine, if we want to do this way then we will get a producer and use it all the way in the test.

**Review by majisourav99 — APPROVED** [review]
LGTM! Thanks Jialin

**Review by gaojieliu — DISMISSED** [review]

**Review by majisourav99 — APPROVED** [review]

**Review by adamxchen — APPROVED** [review]
Thanks for adding more unit test coverage!

**Pull request merged**